lib/common.js

```diff
@@ -1,35 +1,68 @@
 var MysqlTypes = {
-  'DECIMAL': 0,
-  'TINY': 1,
-  'SHORT': 2,
-  'LONG': 3,
-  'FLOAT': 4,
-  'DOUBLE': 5,
-  'NULL': 6,
-  'TIMESTAMP': 7,
-  'LONGLONG': 8,
-  'INT24': 9,
-  'DATE': 10,
-  'TIME': 11,
-  'DATETIME': 12,
-  'YEAR': 13,
-  'NEWDATE': 14,
-  'VARCHAR': 15,
-  'BIT': 16,
+  DECIMAL: 0,
+  TINY: 1,
+  SHORT: 2,
+  LONG: 3,
+  FLOAT: 4,
+  DOUBLE: 5,
+  NULL: 6,
+  TIMESTAMP: 7,
+  LONGLONG: 8,
+  INT24: 9,
+  DATE: 10,
+  TIME: 11,
+  DATETIME: 12,
+  YEAR: 13,
+  NEWDATE: 14,
+  VARCHAR: 15,
+  BIT: 16,
   // Fractional temporal types in MySQL >=5.6.4
-  'TIMESTAMP2': 17,
-  'DATETIME2': 18,
-  'TIME2': 19,
-  'NEWDECIMAL': 246,
-  'ENUM': 247,
-  'SET': 248,
-  'TINY_BLOB': 249,
-  'MEDIUM_BLOB': 250,
-  'LONG_BLOB': 251,
-  'BLOB': 252,
-  'VAR_STRING': 253,
-  'STRING': 254,
-  'GEOMETRY': 255,
+  TIMESTAMP2: 17,
+  DATETIME2: 18,
+  TIME2: 19,
+  NEWDECIMAL: 246,
+  ENUM: 247,
+  SET: 248,
+  TINY_BLOB: 249,
+  MEDIUM_BLOB: 250,
+  LONG_BLOB: 251,
+  BLOB: 252,
+  VAR_STRING: 253,
+  STRING: 254,
+  GEOMETRY: 255
+};
+
+var MysqlCodeTypes = {
```

Comment:
Instead of writing this out hardcoded twice, I think it would be better to generate the second object.

```javascript
// Provide flipped object for quick look-up
var MysqlCodeTypes = {}
Object.keys(MysqlTypes).forEach(function(typeId) {
  MysqlCodeTypes[MysqlTypes[typeId]] = typeId;
});
```